scripts/generator/graphql_generator/csharp/SDK/Android/AndroidNativeCheckout.cs.erb

```diff
@@ -0,0 +1,78 @@
+#if UNITY_ANDROID
+namespace <%= namespace %>.SDK.Android {
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Runtime.InteropServices;
+    using <%= namespace %>.SDK;
+    using <%= namespace %>.MiniJSON;
+
+    #if !SHOPIFY_MONO_UNIT_TEST
+    using UnityEngine;
+    #endif
+
+    public class AndroidNativeCheckout : INativeCheckout {
+        private CartState CartState;
+        private AndroidJavaObject AndroidPayCheckoutSession;
+
+        private CheckoutSuccessCallback OnSuccess;
+        private CheckoutCancelCallback OnCancelled;
+        private CheckoutFailureCallback OnFailure;
+
+        public AndroidNativeCheckout(CartState cartState) {
+            CartState = cartState;
+        }
+
+        public bool CanShowPaymentSetup() {
+            // TODO
+            return true;
+        }
+
+        public void ShowPaymentSetup() {
+            // TODO
+        }
+
+        public bool CanCheckout(PaymentSettings paymentSettings) {
+            // TODO: Invoke methods in the plugin to determine if we can perform Android Pay
+            return true;
+        }
+
+        public void Checkout(
+            string key, 
+            PaymentSettings paymentSettings, 
+            CheckoutSuccessCallback success, 
+            CheckoutCancelCallback cancelled, 
+            CheckoutFailureCallback failure
+        ) {
+            // TODO: Store callbacks and extract items we need from the cart to pass to Android Pay.
+            OnSuccess = success;
+            OnCancelled = cancelled;
+            OnFailure = failure;
+
+            var checkout = CartState.CurrentCheckout;
+
+            // TODO: Replace with Shop name
+            var merchantName = "mables-cables";
```

Comment:
See https://github.com/Shopify/unity-buy-sdk/pull/233 to get the actual shop name